strawberryfield.module

```diff
@@ -74,109 +71,81 @@ function strawberryfield_node_presave(ContentEntityInterface $entity) {
+    // @TODO use updated $event object to debug things if an event failed.
+    // via $event->getProcessedBy();
   }
 }
+
 /**
  * Implements hook_ENTITY_TYPE_revision_create().
- *
- * {@inheritdoc}
  */
 function strawberryfield_node_revision_create(ContentEntityInterface $new_revision, ContentEntityInterface $entity, $keep_untranslatable_fields) {
-
   if ($sbf_fields = \Drupal::service('strawberryfield.utility')->bearsStrawberryfield($entity)) {
     $event_type = StrawberryfieldEventType::NEW_REVISION;
     $event = new StrawberryfieldCrudEvent($event_type, $entity, $sbf_fields);
-    /** @var \Symfony\Component\EventDispatcher\EventDispatcher $dispatcher */
-    $dispatcher = \Drupal::service('event_dispatcher');
-    $dispatcher->dispatch($event_type, $event);
+    \Drupal::service('event_dispatcher')->dispatch($event_type, $event);
   }
 }
 
 /**
  * Implements hook_ENTITY_TYPE_delete().
- *
- * {@inheritdoc}
  */
 function strawberryfield_node_delete(ContentEntityInterface $entity) {
-
   if ($sbf_fields = \Drupal::service('strawberryfield.utility')->bearsStrawberryfield($entity)) {
     $event_type = StrawberryfieldEventType::DELETE;
     $event = new StrawberryfieldCrudEvent($event_type, $entity, $sbf_fields);
-    /** @var \Symfony\Component\EventDispatcher\EventDispatcher $dispatcher */
-    $dispatcher = \Drupal::service('event_dispatcher');
-    $dispatcher->dispatch($event_type, $event);
+    \Drupal::service('event_dispatcher')->dispatch($event_type, $event);
   }
-
 }
 
 /**
  * Implements hook_ENTITY_TYPE_revision_delete().
- *
- * {@inheritdoc}
  */
 function strawberryfield_node_revision_delete(ContentEntityInterface $entity) {
 
   if ($sbf_fields = \Drupal::service('strawberryfield.utility')->bearsStrawberryfield($entity)) {
     $event_type = StrawberryfieldEventType::DELETE;
     $event = new StrawberryfieldCrudEvent($event_type, $entity, $sbf_fields);
-    /** @var \Symfony\Component\EventDispatcher\EventDispatcher $dispatcher */
-    $dispatcher = \Drupal::service('event_dispatcher');
-    $dispatcher->dispatch($event_type, $event);
+    \Drupal::service('event_dispatcher')->dispatch($event_type, $event);
   }
-  //@TODO move this to an event subscriber.
+  // @TODO move this to an event subscriber.
   strawberryfield_invalidate_fieldefinition_caches($entity);
 }
 
-
 /**
  * Invalidate the cache for strawberryfields field type defintions.
  *
- * @param \Drupal\node\NodeInterface $node
+ * @param \Drupal\Core\Entity\ContentEntityInterface $entity
+ *   The entity to invalidate cache.
  */
 function strawberryfield_invalidate_fieldefinition_caches(ContentEntityInterface $entity) {
-
-  //@TODO do the same for StrawberryfieldKeyNameProvider Plugins
-  if ($entity->isPublished() && $entity->isDefaultRevision()) {
+  // @TODO do the same for StrawberryfieldKeyNameProvider Plugins
+  if ($entity instanceof EntityPublishedInterface && $entity->isPublished() && $entity->isDefaultRevision()) {
```

Comment:
You're basically calling `isPublished` on entity but for the entity to have `isPublished` method, it needs to implement EntityPublishedInterface
If we're checking nodes only, let's check for nodes only :)

Reply:
True, again, this is right and I'm not asking for a change. I'm just saying that some entities may use a trait instead of extending the interface. A simple in code comment saying that is sufficient or explaining this change in the Pull Comment. Pull comments are very helpful.

Reply:
If entities implement EntityPublishedTrait without implementing the right interface they need to fix :) IMHO

Reply:
CommentInterface does implement EntityPublishedInterface, I think core is pretty safe territory. Just updated the pull comment.

Reply:
True! Thanks for correcting me.